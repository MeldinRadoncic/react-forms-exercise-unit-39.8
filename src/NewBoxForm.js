import React, {useState} from "react";
import {v4 as uuid} from 'uuid'



const NewBoxForm = ({addBox}) => {
    const INITIAL_STATE = {
color:'',
width:'',
height:''
    }
const [formData,setFormData] = useState(INITIAL_STATE)

const handleSubmit = (e) => {
    e.preventDefault();
    const {color,width,height} = formData;
    if(color !== '' && width !=='' && height !== '') {
        addBox({...formData, id:uuid()})
        setFormData({color:'',width:'',height:''})
    }else {
    alert('Inputs cannot be empty!')
    }
    
}

const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(data => ({
      ...data,
      [name]: value
    }))}


return (
    <div>
        <form onSubmit={handleSubmit} >
            <label htmlFor='color'>Color</label>
            <input 
            type='text'
            placeholder="Enter a color..."
            name="color"
            id="color"
            value={formData.color}
            onChange={handleChange}
            />

<label htmlFor='width'>Width</label>
            <input 
            type='text'
            placeholder="Enter a width..."
            name="width"
            id="width"
            value={formData.width}
            onChange={handleChange}
            />

<label htmlFor='height'>Height</label>
            <input 
            type='text'
            placeholder="Enter a height..."
            name="height"
            id="height"
            value={formData.height}
            onChange={handleChange}
            />

            <button>Add Box</button>

        </form>
    </div>
)

}



export default NewBoxForm;