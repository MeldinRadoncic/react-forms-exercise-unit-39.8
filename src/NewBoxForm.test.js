import { render } from "@testing-library/react";
import React from "react";
import NewBoxForm from './NewBoxForm'

it('Renders without crashing', () => {
    render(<NewBoxForm/>)
})

it('Matches snapshot', () => {
    const {asFragment} = render(<NewBoxForm/>)
    expect(asFragment()).toMatchSnapshot();
})