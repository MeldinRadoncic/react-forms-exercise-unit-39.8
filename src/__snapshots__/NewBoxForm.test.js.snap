// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        placeholder="Enter a color..."
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        placeholder="Enter a width..."
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        placeholder="Enter a height..."
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
